### ISSUE TITLE
Binder not working from Harvard Dataverse

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse.harvard.edu#328]
For now, a couple links:

- discussion with Binder team: https://jupyter.zulipchat.com/#narrow/channel/103349-ask-anything/topic/Binder.20Dataverse.20error/near/476223581
- debugging: https://github.com/IQSS/repo2docker/commit/1e0d72629a720416382e5a9359812713de9d766f

**Comment by pdurbin** [comment]
Two updates:

- short term: we have [asked](https://iqss.slack.com/archives/C01BY1TP4H4/p1734363639050469?thread_ts=1729103069.613969&cid=C01BY1TP4H4) for `/citation` to be unblocked
- long term: simplifying repo2docker to only operate on datasets should mean we can re-block `/citation`:
  - https://github.com/yuvipanda/repo2docker/pull/7

**Comment by pdurbin** [comment]
Unblocking `/citation` seemed to work. 🎉  https://mybinder.org/v2/dataverse/10.7910/DVN/TJCLKP/ (which is where the Binder button sends you, for example) opens up my dataset:

![Screenshot 2024-12-16 at 11 29 35 AM](https://github.com/user-attachments/assets/7866754d-a3fc-4dcb-aa5d-be406127cc38)

For a longer term fix I'll keep monitoring my PR at https://github.com/yuvipanda/repo2docker/pull/7 which I announced at https://jupyter.zulipchat.com/#narrow/channel/103349-ask-anything/topic/Binder.20Dataverse.20error/near/489270587

**Comment by pdurbin** [comment]
This is the new PR to track:

- https://github.com/jupyterhub/repo2docker/pull/1390

Once it's merged and deployed, we can try pulling the block back in place for `/citation`.

While we're bothering those folks, we'll unblock `/openapi`, which was asked about at https://jupyter.zulipchat.com/#narrow/channel/103349-ask-anything/topic/Binder.20Dataverse.20error/near/489373873

**Comment by pdurbin** [comment]
@yuvipanda now that this PR has been merged...

- https://github.com/jupyterhub/repo2docker/pull/1390

... do you think we should try putting the block back in place for `/citation`?

**Comment by pdurbin** [comment]
Following some [discussion](https://jupyter.zulipchat.com/#narrow/channel/103349-ask-anything/topic/Binder.20Dataverse.20error/near/495526216), we'll track the deployment of the fix above with with this issue:

- https://github.com/jupyterhub/mybinder.org-deploy/issues/3191

**Comment by pdurbin** [comment]
Per https://github.com/jupyterhub/mybinder.org-deploy/issues/3191 the deployment was done on Dec 21, 2024.

But let's not change anything on a Friday. Maybe next week.

As a reminder we're asking for two things:

- put the block back in place for `/citation`
- remove the block from `/openapi` (while we're asking for changes anyway)

**Comment by pdurbin** [comment]
@landreev said he's coordinate with LTS. Thanks!

**Comment by pdurbin** [comment]
> put the block back in place for `/citation`

Done. And I re-tested Binder. Works fine. I gave a [shout out](https://jupyter.zulipchat.com/#narrow/channel/103349-ask-anything/topic/Binder.20Dataverse.20error/near/508105755) to the developer who helped us out. ❤

**Comment by yuvipanda** [comment]
\o/

This is also a perfect good example of open source and open infrastructure working well - this led to big improvements in repo2docker and binderhub, so yay :) Thank you for reporting and engaging with us throughout this process!
